applications/launchpad_v2/src/components/Tag/index.tsx

```diff
@@ -0,0 +1,102 @@
+import { useContext } from 'react'
+import { ThemeContext, CSSProperties } from 'styled-components'
+
+import Text from '../Text'
+
+import { TagProps } from './types'
+
+import { TagContainer, IconWrapper } from './styles'
+
+/**
+ * @name Tag
+ * @typedef TagProps
+ *
+ * @prop {ReactNode} [children] - text content to display
+ * @prop {CSSProperties} [style] - optional component styles
+ * @prop {'blue' | 'running' | 'warning' | 'expert'} [type] - tag types to determine color settings
+ * @prop {ReactNode} [icon] - optional SVG icon
+ * @prop {ReactNode} [subText] - optional additional tag text
+ *
+ * @example
+ * <Tag type='running' style={extraStyles} icon={<someIconComponent/>} subText='Mainnet'>
+ *    Running
+ * </Tag
+ */
+
+const Tag = ({
+  children,
+  style,
+  type,
+  variant = 'small',
+  icon,
+  subText,
+}: TagProps) => {
+  const theme = useContext(ThemeContext)
+
+  let baseStyle: CSSProperties = {}
+  let textStyle: CSSProperties = {}
+
+  switch (type) {
+    case 'blue':
+      baseStyle = {
+        backgroundColor: theme.info,
+        color: theme.infoText,
+      }
+      break
+    case 'running':
+      baseStyle = {
+        backgroundColor: theme.on,
+        color: theme.onText,
+      }
+      break
+    case 'warning':
+      baseStyle = {
+        backgroundColor: theme.warning,
+        color: theme.warningText,
+      }
+      break
+    case 'expert':
+      baseStyle = {
+        backgroundColor: theme.expert,
+      }
+      textStyle = {
+        backgroundImage: theme.expertText,
+        WebkitBackgroundClip: 'text',
+        color: 'transparent',
+      }
+  }
```

Comment:
I'm wondering if we need some default style for the Tag component, bc if we do not specify `type`, the Tag has transparent background. So maybe we could add the `default` statement to the `switch-case` block that:

a) adds some grayish background
b) ...or we use the `blue` type as default.

I would go with option (b). So then you can remove the `case 'blue':` and move its content the default block.

Reply:
👍